src/game.cpp

```diff
@@ -11483,7 +11483,15 @@ void game::reload( int pos )
     }
 
     auto loc = it->pick_reload_ammo( u, true );
-    if( loc.get_item() ) {
+
+    const auto ammo = loc.get_item();
+    if( ammo ) {
+        if( ammo->is_magazine() && ammo->ammo_remaining() == 0 ) {
+            if( !query_yn( _( "Reload using an empty magazine?" ) ) ) {
```

Comment:
`RELOAD_ONE` could be special cased.

Though maybe the always menu thing may not be a good idea. Well, as long as it isn't an alternation between menu/query depending on number of magazines, it should be OK. It could be a tiny bit risky when your only magazine is huge, barely filled or otherwise "bad", but always having a menu could be inconvenient.